### ISSUE TITLE
Vala: resolved names of pc files don't lead to right --pkg name.

### ISSUE THREAD

**Issue by mdickie** [mesonbuild/meson#687]
```
project('ggit-test', ['vala', 'c'])

glib = dependency('glib-2.0')
gobject = dependency('gobject-2.0')
git = dependency('libgit2-glib-1.0')

# For the git dependency vala argument should be --pkg ggit-1.0, otherwise 
# it won't compile. It would be possible to simply add that arg to executable but
# still it won't compile because you need to remove that arg --pkg libgit2-glib-1.0

executable('ggit-test', 'main.vala',
           dependencies : [glib, gobject, git])
```

**Comment by jpakkane** [comment]
Given a pkgconfig dependency `foo` how would you determine what the proper Vala pkgname should be?

**Comment by mdickie** [comment]
The thing is the problem is twofold: The Vala Compiler needs to know where

**Comment by mdickie** [comment]
The most basic problem is probably that vapi files (--pkg flags direct to them) don't give hints, on which pkg-config files are required by that api. In any case that wouldn't be needed by Autotools because you set pkg-config libraries and --pkg flags on your own. So the solution could be to have library with two different names (pkg-config name, vapi name) but I think that's too complicated. However, in my opinion this could be solved more intuitively by asking gnome developers to add hints in vapi files in order to find the correct pkg-config file.

My example would look then like this:
```
project('ggit-test', ['vala', 'c'])

glib = dependency('glib-2.0')
gobject = dependency('gobject-2.0')
git = dependency('git-1.0')

executable('ggit-test', 'main.vala',
           dependencies : [glib, gobject, git])
```

**Comment by astavale** [comment]
Your VAPI should have the same name as the pkg-config file, minus the suffixes. For example the VAPI for the glib-2.0.pc file is referenced as --pkg glib-2.0 There may be some confusion here because the Vala compiler automatically includes --pkg glib-2.0 and --pkg gobject-2.0, even when they are not specified on the command line. Another example would be for Gtk+ 3. The .pc file is gtk+-3.0.pc so to get the flags for the C compiler use `pkg-config --cflags --libs gtk+-3.0` and to reference the VAPI pass `--pkg gtk+-3.0` to the Vala compiler. The Vala compiler will pass the relevant `pkg-config` command to the C compiler if you are building the binary as part of the `valac` command.

See [Legacy Bindings - The VAPI File](https://wiki.gnome.org/Projects/Vala/LegacyBindings#The_VAPI_File) for more details.

**Comment by mdickie** [comment]
Ok, I passed my issue to gnome bugzilla: https://bugzilla.gnome.org/show_bug.cgi?id=770910

**Comment by nirbheek** [comment]
This is fixed with the implementation of #1023.
